test/node/medtronic/testMedtronicSimulator.js

```diff
@@ -0,0 +1,574 @@
+        simulator.basal(tempBasal);
+        simulator.basal(basal2);
+        expect(simulator.getEvents()).deep.equals([
+          basal1.done(),
+          expectedTempBasal
+        ]);
+      });
+
+      it('checks for schedule change', function() {
+
+        tempBasal.jsDate = new Date(tempBasal.deviceTime);
+
+        simulator.pumpSettings(settings);
+        simulator.basal(basal1);
+        simulator.basal(tempBasal);
+        simulator.basal(basal2);
+
+        var expectedTempBasal1 = _.cloneDeep(tempBasal.done());
+        expectedTempBasal1.suppressed.rate = 1.3;
+        expectedTempBasal1.duration = 50000;
+        expectedTempBasal1.payload.duration = 1800000;
+        delete expectedTempBasal1.index;
+        delete expectedTempBasal1.jsDate;
+
+        var expectedTempBasal2 = _.cloneDeep(expectedTempBasal1);
+        expectedTempBasal2.clockDriftOffset = 0;
+        expectedTempBasal2.duration = 1750000;
+        expectedTempBasal2.time = '2014-09-25T17:10:50.000Z';
+        expectedTempBasal2.deviceTime = '2014-09-25T18:10:50';
+        expectedTempBasal2.annotations = [{code: 'medtronic/basal/fabricated-from-schedule'}];
+        expectedTempBasal2.suppressed.rate = 0.475;
+        delete expectedTempBasal2.payload.duration;
+
+        delete basal1.index;
+        delete basal1.jsDate;
+
+        expect(simulator.getEvents()).deep.equals([
+          settings,
+          basal1.done(),
+          expectedTempBasal1,
+          expectedTempBasal2
+        ]);
+      });
+
+      it('checks for schedule change over midnight', function() {
+
+        simulator.pumpSettings(settings);
+        simulator.basal(basal1);
+        simulator.basal(tempBasalOverMidnight);
+        simulator.basal(basal2);
+
+        var expectedTempBasal1 = _.cloneDeep(tempBasalOverMidnight.done());
+        expectedTempBasal1.suppressed.rate = 1.3;
+        expectedTempBasal1.duration = 3000000;
+        expectedTempBasal1.payload.duration = 7200000;
+        delete expectedTempBasal1.index;
+        delete expectedTempBasal1.jsDate;
+
+        var expectedTempBasal2 = _.cloneDeep(expectedTempBasal1);
+        expectedTempBasal2.clockDriftOffset = 0;
+        expectedTempBasal2.duration = 4200000;
+        expectedTempBasal2.time = '2014-09-25T23:00:00.000Z';
+        expectedTempBasal2.deviceTime = '2014-09-26T00:00:00';
+        expectedTempBasal2.annotations = [{code: 'medtronic/basal/fabricated-from-schedule'}];
+        expectedTempBasal2.suppressed.rate = 0.2;
+        delete expectedTempBasal2.payload.duration;
+
+        delete basal1.index;
+        delete basal1.jsDate;
+
+        expect(simulator.getEvents()).deep.equals([
+          settings,
+          basal1.done(),
+          expectedTempBasal1,
+          expectedTempBasal2
+        ]);
+      });
+
+      it('checks temp basal over midnight without schedule change', function() {
+
+        settings.basalSchedules.standard[0].start = 7200000; // only change at 2am
```

Comment:
Yep, total sense. I've updated the test which then showed that I should be checking that the rate actually changes.